src/index.coffee

```diff
@@ -6,45 +6,39 @@ BrowserTransport = require './transports/browser-transport'
 CliTransport = require './transports/cli-transport'
 Logger = require './logger'
 
-
-getGlobalScope = ->
-  if typeof global != 'undefined'
-    return global
-
-  return window
-
-
-isBrowser = (globalScope) ->
-  return (
-    globalScope.window?.document?.createElement and
-    globalScope.window?.name != 'nodejs'
-  )
+{ getGlobalScope, isBrowser } = require './utils'
+{ setupTransport } = require './setup-transport'
 
 
 module.exports =
   LogLevels: LogLevels
   
   Logger: Logger
 
+  BrowserTransport: BrowserTransport
+  CliTransport: CliTransport
+
   create: (options = {}) ->
     globalScope = getGlobalScope()
     browser = isBrowser(globalScope)
 
     if browser
-      transport = new BrowserTransport(globalScope.console)
-      transport.setLogLevelPrefixes(prefixes.browser)
-      transport.setLogLevelStyles(colors.browser)
+      output = globalScope.console
+      transportSource = BrowserTransport
+
     else
-      transport = new CliTransport(globalScope.process.stderr)
-      transport.setLogLevelPrefixes(prefixes.cli)
-      transport.setLogLevelColors(colors.cli)
+      output = globalScope.process.stderr
+      transportSource = CliTransport 
 
-    logger = new Logger(transport, options)
+    transport = if options.transport then new options.transport(output) else new transportSource(output)
+    transportWithSetup = setupTransport(transport)
```

Comment:
It would made a lot of sense to me to have this setup as a part of the transport factory. There can be `createBrowserTransport()` and `createCliTransport()` factories or something similar which would only be used in case no custom transport is passed in. As I mentioned in the comment in `Logger#switchTransport()`, the custom transport should be fully configured before being passed in here.